### PR TITLE
#506: Fig bug validate reference-tx-hashes for user content transaction

### PR DESCRIPTION
**Pull request by hieudinhngoc-ncc**
# Ticket
#506 

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- This is fixbug task

Design link (diagram / doc):
- This is fixbug task

# Testing
- [x] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: ledger/ledger.go [hunk 1]
```diff
@@ -312,6 +312,20 @@ func (l *Ledger) validateUserContent(tx *transaction.Transaction, hashRelatedCon
...
+	// Validate reference transactions
+	if len(content.ReferenceTxHashes) > 0 {
+		if len(content.ReferenceTxHashes) > validation.MaxReferenceTxs {
+			return fmt.Errorf("user content data is invalid")
+		}
+
+		for _, refTxHash := range content.ReferenceTxHashes {
+			if _, exists := hashRelatedContentTx[refTxHash]; !exists {
+				return fmt.Errorf("user content data is invalid")
+			}
```
Comment (TrongNgoVan) [inline ledger/ledger.go:324]: tôi có một thắc mắc: 
  1. Đoạn này có vẻ như đang check tx trong mem được map `hashRelatedContentTx[refTxHash]` đúng không ô nhỉ? Nếu vậy sao mình không dùng `getTxByHash` trong `txStore` nhỉ? ( theo như tôi thấy trong code có một số hàm như thế ), vì có vẻ check trong mem ( tuy nhanh hơn nhưng đánh đổi tx đó có vẻ không uy tín lắm đúng k ô nhỉ?)

Comment (hieudinhngoc-ncc) [reply ledger/ledger.go:324]: Nếu mà bạn đọc lên trên 1 chút thì `hashRelatedContentTx[refTxHash]` nó được lấy từ store ra theo `getBatch` ấy k dùng `getTxByHash` ở đây vì nếu for thì mình sẽ phải hit DB nhiều => thay vì thế nên tôi lấy hết tx ra ở trên rồi. Bản chất cũng là lấy từ store ra thôi.

Comment (TrongNgoVan) [reply ledger/ledger.go:324]: > Nếu mà bạn đọc lên trên 1 chút thì `hashRelatedContentTx[refTxHash]` nó được lấy từ store ra theo `getBatch` ấy k dùng `getTxByHash` ở đây vì nếu for thì mình sẽ phải hit DB nhiều => thay vì thế nên tôi lấy hết tx ra ở trên rồi. Bản chất cũng là lấy từ store ra thôi.

confirm

**Review by TrongNgoVan — APPROVED** [review]

**Review by hoangdohuy — APPROVED** [review]
LGTM
